runtime/sema/check_function.go

```diff
@@ -187,6 +187,7 @@ func (checker *Checker) checkFunction(
 			functionActivation.InitializationInfo = initializationInfo
 
 			if functionBlock != nil {
+				oldMappedAccess := checker.entitlementMappingInScope
```

Comment:
Maybe move the code starting here and ending with the new reset (L203) into an immediately invoked function, and move the reset line (203) below this line, as a `defer` statement. That makes the scope more obvious and keeps setting and resetting closer together.